src/datagrid/DataGrid.tsx

```diff
@@ -1146,6 +1152,8 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
                                     const showColumn =
                                         this.isDetailPaneOpen() &&
                                         column.columnID !== this.getFirstVisibleColumn()!.columnID
+                                            ? false
+                                            : column.isHidden
```

Comment:
instate of this can we modify getFirstVisibleColumn() ?

Reply:
simplified this